python/pyramid/server.py

```diff
@@ -0,0 +1,22 @@
+from wsgiref.simple_server import make_server
+from pyramid.config import Configurator
+from pyramid.response import Response
+from pyramid.response import Response
+
+
+def empty(request):
+    return Response()
+
+
+def show(request):
+    return Response(request.matchdict["id"])
+
+
+with Configurator() as config:
+    config.add_route("index", "/")
+    config.add_route("create_user", "/user", request_method="POST")
+    config.add_route("get_user", "/user/{id}")
+    config.add_view(empty, route_name="index")
+    config.add_view(empty, route_name="create_user")
+    config.add_view(show, route_name="get_user")
+    app = config.make_wsgi_app()
```

Comment:
I'm not sure why you omitted a few lines that typically exist in the simplest Pyramid app, maybe because you use gunicorn for benchmarking? If yours does not run, then try the following.

```python
if __name__ == '__main__':
    with Configurator() as config:
        config.add_route("index", "/")
        config.add_route("create_user", "/user", request_method="POST")
        config.add_route("get_user", "/user/{id}")
        config.add_view(empty, route_name="index")
        config.add_view(empty, route_name="create_user")
        config.add_view(show, route_name="get_user")
        app = config.make_wsgi_app()
    server = make_server('0.0.0.0', 6543, app)
    server.serve_forever()
```

Reply:
Yes, I need `gunicorn` to run this, of course I'm open to suggestions :heart: 

I've not seed any **recommended** way in the documentation

Reply:
Pyramid has no opinion about the web server you use in front of your app. You will not find a recommended way.

The [Pyramid Community Cookbook offers a suggestion for deploying with gunicorn](https://docs.pylonsproject.org/projects/pyramid_cookbook/en/latest/deployment/gunicorn.html). This is not officially supported, but something the community contributed. I think it is outdated because it suggests using `paste` to start serving a Pyramid app. `pserve` is more common usage.

You will see that your minimalist app uses Python's [`wsgiref.simple_server`](https://docs.python.org/3/library/wsgiref.html#module-wsgiref.simple_server) to serve the app. If you want to use gunicorn, you will need to make some modifications.

The easiest and officially supported way to get a working Pyramid app is to start with the [pyramid-cookiecutter-starter](https://github.com/Pylons/pyramid-cookiecutter-starter) to generate the app, then follow the instructions of the generated `README.txt` to install your app and its dependencies, and to run it. You could compare what is generated from that, and modify your one-file app into a distributable package accordingly.

Our Quick Tutorial includes one step [03: Application Configuration with `.ini` Files](https://docs.pylonsproject.org/projects/pyramid/en/latest/quick_tutorial/ini.html) that demonstrates and explains how this all works, with further references to the documentation.